lib/plain/plain-client.ts

```diff
@@ -0,0 +1,34 @@
+import { createCMAHttpClient, ClientParams } from '../create-cma-http-client'
+import * as endpoints from './index'
+import { AxiosInstance } from 'axios'
+
+const withHttp = <T extends any[], R>(
+  http: AxiosInstance,
+  fn: (http: AxiosInstance, ...rest: T) => R
+) => {
+  return (...rest: T) => fn(http, ...rest)
+}
+
+export const createPlainClient = (params: ClientParams) => {
```

Comment:
What about:

```

type EndpointMethod<T extends any[], R> = (http: AxiosInstance, ...rest: T) => R;
type EndpointMethodDict<T extends any[], R> = Record<string, EndpointMethod<T, R>>;

const withHttp = <T extends any[], R>(http: AxiosInstance, fn: EndpointMethod<T, R>) => (...rest: T) => fn(http, ...rest)

const fromEndpoints = <T extends any[], R>(http: AxiosInstance, endpoints: EndpointMethodDict<T, R>) => {
  return Object.entries(endpoints)
    .reduce(
      (endpointsWithHttp, [methodName, method]) => ({ ...endpointsWithHttp, [methodName]: withHttp(http, method) }),
      {},
    );
};

export const createPlainClient = (params: ClientParams) => {
  const http = createCMAHttpClient(params)

  return {
    space: fromEndpoints(http, endpoints.space),
    environment: fromEndpoints(http, endpoints.environment),
    contentType: fromEndpoints(http, endpoints.contentType),
    user: fromEndpoints(http, endpoints.user),
    entry: fromEndpoints(http, endpoints.entry),
  }
}
```

Reply:
Awesome if all types work as expected. I'll check it out.